### DISCUSSION TITLE
Add support for Tilt

### DISCUSSION THREAD

**Discussion by ianks** [bootstrap-email/bootstrap-email#110] [Ideas]
[Tilt](https://github.com/rtomayko/tilt) is a thin wrapper over various different template engines. By adding support for Tilt, support for apps built with Hanami, Roda, and other frameworks. Tilt also provides a few things that could be beneficial for performance as well, such as the ability to precompile templates.

I think the adapter could live in this repo, so users of Tilt could simply `require 'bootstrap-email/tilt`. Let me know your thoughts! Love this gem and it this feature would allow us to use it in prod. Cheers!

**Comment by ianks** [comment]
Here's a demo implementation:

```ruby
module Tilt
  class BootstrapEmail < ERBTemplate
    self.default_mime_type = 'text/html'

    def prepare
      super
      require 'bootstrap-email'
    end

    def evaluate(_scope, _locals, &_block)
      ::BootstrapEmail::Compiler.new(super, type: :string).perform_full_compile
    end

    def allows_script?
      false
    end
  end
end
```

**Comment by stuyam** [comment]
Cool! I don't know much about Tilt but that looks interesting. Im open to pull requests adding support for Tilt 👍
